src/lazy/text/value.rs

```diff
@@ -0,0 +1,66 @@
+use crate::lazy::decoder::private::LazyRawValuePrivate;
+use crate::lazy::decoder::{LazyDecoder, LazyRawValue};
+use crate::lazy::encoding::TextEncoding;
+use crate::lazy::raw_value_ref::RawValueRef;
+use crate::lazy::text::buffer::TextBufferView;
+use crate::lazy::text::encoded_value::EncodedTextValue;
+use crate::lazy::text::matched::MatchedValue;
+use crate::{IonResult, IonType, RawSymbolTokenRef};
+use std::fmt;
+use std::fmt::{Debug, Formatter};
+
+/// A value that has been identified in the text input stream but whose data has not yet been read.
+///
+/// If only part of the value is in the input buffer, calls to [`LazyRawTextValue::read`] (which examines
+/// bytes beyond the value's header) may return [`IonError::Incomplete`](crate::result::IonError::Incomplete).
```

Comment:
🗺️ I have a lingering reference to "the value's header" that was a copy/paste error from this type's binary counterpart. I'll fix that before merging.